---
title: "SARS-CoV-2 NPC response"
subtitle: "Testing response to perturbing IL10RB"
author:
- name: "[Gabriel Hoffman](http://gabrielhoffman.github.io)"
  affiliation: | 
    Icahn School of Medicine at Mount Sinai, New York
date: "Run on `r Sys.time()`"
documentclass: article
output:
  html_document:
    toc: true
    toc_float: true
params:
  upload: FALSE
---


<!---

cd /Users/gabrielhoffman/workspace/repos/sars_cov_2_npc_response/
# rm -rf analysis_cache/

rmarkdown::render("analysis.Rmd")


scp analysis.html sklar1:/hpc/users/hoffmg01/www/software/sars_cov_2_npc_response



--->
 


```{r knit2synapse, eval= FALSE}
library(synapser)
library(knit2synapse)
synLogin()
knit2synapse::createAndKnitToFolderEntity(file = "combined_analysis/combine_all_DLPFC.Rmd",
                                          parentId ="syn22416298",
                                          folderName = 'Joint analysis of DLPFC count data',
                                          overwrite=TRUE, 
                                          knitmd=TRUE)
```


```{r synapse.parameters, include=FALSE, cache=TRUE, eval=FALSE}

library(githubr)
parentId = 'syn24172641';
activityName = 'Joint analysis of DLPFC count data';
activityDescription = 'Joint analysis of DLPFC count data';

thisFileName <- 'combine_all_DLPFC.Rmd'

# Github link
thisRepo <- getRepo(repository = "CommonMindConsortium/covarr-de", ref="branch", refName='master')
thisFile <- getPermlink(repository = thisRepo, repositoryPath=paste0('combined_analysis/',thisFileName))
```




```{r setup, include=FALSE}
suppressPackageStartupMessages({
library(synapser)

synLogin()

library(ggplot2)
library(dplyr)
library(readxl)
library(gtools)
library(pvclust)
library(data.table)
library(cowplot)
library(EnrichmentBrowser)
library(GSEABase)
library(cowplot)
library(kableExtra)
library(tidyverse)
library(githubr)
library(zenith)
library(knitr)
library(BiocParallel)
library(limma)
library(edgeR)
library(variancePartition)

CODE <- Folder(name = "results", parentId = "syn24182416")
CODE <- synStore(CODE)

storeData = function( name, description){
  file = paste0(name, ".RDS")
  saveRDS( eval(parse(text=name)), file=file)

  nEXP_OBJ = File(file, 
                  name = description, 
                  parentId = CODE$properties$id)
  synStore(nEXP_OBJ)#, used = ALL_USED_IDs, executed = thisFile)
}

source("/Users/gabrielhoffman/workspace/repos/CMC/covarr-de/common_functions/common_functions.R")

}) 
knitr::opts_chunk$set(
  echo = FALSE,
  warning=FALSE,
  message=FALSE,
  error = FALSE,
  tidy = FALSE,
  dev = c('png', 'pdf'),
  cache = TRUE)
```

# Read data
```{r read.data}
# sample metadata
df_meta = read_xlsx( synGet('syn24182417')$path, skip=1 )
df_meta = data.frame(df_meta[,-1])
colnames(df_meta)[colnames(df_meta) == "ID."] = "ID"
df_meta$Gene[df_meta$Gene == "Scramble"] = paste(df_meta$Gene[df_meta$Gene == "Scramble"], df_meta$Method[df_meta$Gene == "Scramble"] )

# virus quantification
df_tax = read.table( synGet('syn24183998')$path, sep=',', header=TRUE)
df_meta = merge(df_meta, df_tax, by.x="ID", by.y="Sample")

# Ampliseq data
df_ampliseq = read_xlsx( synGet('syn24185208')$path, skip=0 )
colnames(df_ampliseq)[colnames(df_ampliseq)=='Sample'] = "ID"
df_ampliseq$ID = gsub("-", "_", df_ampliseq$ID)
df_ampliseq$SeqMethod = 'Complete'
df_ampliseq$SeqMethod[grep("_ampliseq", df_ampliseq$ID)] = "Ampliseq"
df_ampliseq$ID = gsub("_ampliseq", "", df_ampliseq$ID)

df_ampliseq = df_ampliseq[df_ampliseq$SeqMethod == 'Ampliseq',]
colnames(df_ampliseq)[2:5] = paste0(colnames(df_ampliseq)[2:5], '_ampl')

df_meta = merge(df_meta, df_ampliseq, by="ID", sort=FALSE)

# sample RNA-seq QC metrics
df_qc = read.table( synGet('syn24182502')$path )
rownames(df_qc) = gsub("-", "_", rownames(df_qc))
df_qc$ID = rownames(df_qc)

# read RNA-seq counts
df_counts = fread( synGet('syn24182432')$path)
colnames(df_counts) = gsub("-", "_", colnames(df_counts))
geneCounts = data.frame(df_counts[,-c(1:6)], check.names=FALSE)
rownames(geneCounts) = df_counts$Geneid

# order columns of geneCounts to be same as df_meta
idx = match(df_meta$ID, colnames(geneCounts))
geneCounts = geneCounts[,idx]

# get annotation for each gene
geneInfo = df_counts[,1:6]
setkey(geneInfo, "Geneid")
geneInfo = getGeneSymbol(geneInfo, 'Geneid')
```



# QC metrics
Evaluate basic QC metrics from the BAM files
```{r QC.stats, fig.width=8, fig.height=30}

variables.qc = c('PAIR_TOTAL_READS',
'PAIR_PF_INDEL_RATE',
'PAIR_MEAN_READ_LENGTH',
'PAIR_READS_ALIGNED_IN_PAIRS',
'PAIR_PCT_READS_ALIGNED_IN_PAIRS',
'PAIR_PCT_PF_READS_IMPROPER_PAIRS',
'UNPAIRED_READS_EXAMINED',
'UNMAPPED_READS',
'READ_PAIR_DUPLICATES',
'READ_PAIR_OPTICAL_DUPLICATES',
'PERCENT_DUPLICATION',
'ESTIMATED_LIBRARY_SIZE',
'MEDIAN_INSERT_SIZE',
'READ_PAIRS',
'RIBOSOMAL_BASES',
'CODING_BASES',
'UTR_BASES',
'INTRONIC_BASES',
'INTERGENIC_BASES',
# 'CORRECT_STRAND_READS',
# 'INCORRECT_STRAND_READS',
"PCT_R1_TRANSCRIPT_STRAND_READS" ,           
"PCT_R2_TRANSCRIPT_STRAND_READS" ,           
"PCT_RIBOSOMAL_BASES"           ,            
"PCT_CODING_BASES"  ,                        
"PCT_UTR_BASES"    ,                         
"PCT_INTRONIC_BASES" ,                       
"PCT_INTERGENIC_BASES",                      
"PCT_MRNA_BASES"  ,                          
"PCT_USABLE_BASES"                        
# "PCT_CORRECT_STRAND_READS"
)

variables.meta = c('COVID', 'Method', 'Gene', 'Plate','ng.uL...10', 'A260.280', 'A260.230', 'RIN', 'ng.ul', 'pg.ul', 'ng.uL...16')

# Merge metadata and QC metrics
info = merge(df_meta, df_qc[,variables.qc], by.x='ID', by.y="row.names", sort=FALSE)
rownames(info) = info$ID

figList = lapply( variables.qc, function(x){
  ggplot(info, aes_string(x)) + geom_histogram() + theme_classic() + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), legend.position="bottom") 
})

plot_grid(plotlist = figList, ncol=3)
```

# Evaluate the correlation between QC metrics
```{r qc.corr, fig.height=12, fig.width=12}
form = as.formula(paste('~', paste(c(variables.qc, variables.meta), collapse=' + ')))
C = canCorPairs( form, info)
plotCorrMatrix(C)
```

# Study design

```{r study.design}
xtabs(~Gene + Method, info) %>% kable(caption="Batching") %>% kable_styling(full_width=FALSE)
```

# PCA
Samples separate by RIN, Method (CRISPRa/shRNA) and gene target.  Importantly, there is no separation along the variables of biological interest.  The effect of the first two components can be removed while preserving the biological signal
```{r pca, fig.width=9, fig.height=9}

keep <- filterByExpr(geneCounts)
dge = DGEList( geneCounts[keep,])
dge = calcNormFactors(dge)
geneExpr = cpm(dge, log=TRUE)

# https://www.gtexportal.org/home/faq#tpm
geneExprRPKM = rpkm(dge, gene.length=geneInfo[rownames(dge),Length])
geneExprLogTPM = apply(geneExprRPKM, 2, function(x) log2(x / sum(x) * 1e6 + 0.1))

C <- cor(geneExpr, method="spearman")

hcl = hclust(as.dist(1-C), method='ward.D2')
# plot(hcl)

mds = cmdscale(as.dist(1-C))

df_mds = data.frame(ID=rownames(mds), MDS1 = mds[,1], MDS2 = mds[,2])

info = merge(info, df_mds, by="ID", sort=FALSE)
rownames(info) = info$ID

variables.meta = c('COVID', 'Method', 'Gene', 'Plate')

figList = lapply( variables.meta, function(x){

  ggplot(info, aes(MDS1, MDS2)) + geom_point(aes_string(color=x)) + theme_classic() + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), legend.position="bottom") + scale_color_discrete( name = x)
})
plot_grid( plotlist = figList, ncol=2)
```

### Summary of confounding along MDS

```{r pca.final, fig.width=5, fig.height=5}
ggplot(info, aes(MDS1, MDS2, shape=Method, size=RIN, color=Gene)) + geom_point() + theme_classic() + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), legend.position="right")
```

# Corralates of MDS 2
Since MDS1 correlates with RIN, identify the top covariates correlated with MDS2.
Just comparing IL10RB-1 with Scramble CRISPRa on the subset of samples produces 12K DE genes ( no enrichments).  This comparison accounting for the global RIN trend is ~ 7K genes.  Accounting for MDS produces ~ 2K genes, where the enrichments are qualitatively similar to the previous case.
```{r mds.cor.2}
idx = apply(df_qc, 2, function(x) (var(x)> 0))
idx = names(idx)[idx]
idx = idx[!is.na(idx)]

df2 = merge(df_mds, df_qc[,c("ID",idx)], by="ID")
df = as.data.frame( t(cor(df2$MDS2, df2[,-1])))
df$name = rownames(df)

df[abs(df$V1) > 0.4,1,drop=FALSE] %>% kable(caption="Correlates of MDS2") %>% kable_styling(full_width=FALSE)
```


## SARS.CoV.2 quantification
Quantification of SARS.CoV.2 corresponds to COVID status.  Important, there is a very strong trend between SARS.CoV.2 quantify and RIN on the CoV+ samples.  The interpretation is unclear.
```{r covid.quant, fig.width=5, fig.height=4}
ggplot(df_meta, aes(Gene, SARS.CoV.2, color=COVID)) + geom_point() + theme_classic() + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), legend.position="right", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

ggplot(df_meta, aes(RIN, SARS.CoV.2, color=COVID)) + geom_point() + theme_classic() + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), legend.position="right") + geom_smooth(method='lm')

ggplot(df_meta, aes(`SARS.CoV.2` , `SARS-CoV-2_ampl` / Primate_ampl, color=COVID, shape=Method)) + geom_point() + theme_classic() + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5)) + ggtitle("Complete versus AmpliSeq") #  + geom_smooth(method='lm', se=FALSE)

R.all = with(df_meta, cor(`SARS.CoV.2` , `SARS-CoV-2_ampl` / Primate_ampl, method="spearman"))

R.all.positive = with(df_meta[df_meta$COVID=="CoV+",], cor(`SARS.CoV.2` , `SARS-CoV-2_ampl` / Primate_ampl, method="spearman"))
```

Spearman correlation: `r R.all`. Spearman correlation for CoV+ samples: `r R.all.positive`.




# QC metrics by gene
```{r QC.metrics.by.gene, fig.width=8, fig.height=8}
info %>% 
  select(c('Gene', 'COVID','RIN', 'A260.280', 'A260.230', 'ng.ul', 'pg.ul', 'ng.uL...16', 'PERCENT_DUPLICATION')) %>%
  reshape2::melt(id.vars=c("Gene", "COVID")) %>%
  ggplot(aes(Gene, value, color=COVID)) + geom_point() + facet_wrap(~variable, scales="free_y") + theme_classic() + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), legend.position="bottom", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) 
```

# Visualize effect of shRNA/CRISPR on expression of target gene
```{r summarize.effects, fig.width=12}
tab = getGeneSymbol(geneInfo, 'Geneid')

symbols = c('IL10RB', 'IFNAR2', 'NEUROD2')

gi = data.frame(tab[tab$Symbol %in% symbols,])

df = data.frame(ID = colnames(geneExpr), t(geneExpr[gi$Geneid,]))
# df = data.frame(ID = colnames(geneExprLogTPM), 
    # t(geneExprLogTPM[gi$Geneid,]))

df = reshape2::melt(df, id.vars="ID", variable.name="gene_id", value.name="Expression")

# get gene symbols
df = getGeneSymbol(df, 'gene_id')

# merge with metadata
df = merge(df, info[,1:8], by='ID')

# plot
ggplot(df[df$Method == "shRNA",], aes(Gene, Expression, color=COVID)) + geom_point() + theme_bw(16) + facet_wrap(~Symbol) + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))  + ylab(bquote(Expression~(log[2]~CPM))) + ggtitle("shRNA")

ggplot(df[df$Method == "CRISPRa",], aes(Gene, Expression, color=COVID)) + geom_point() + theme_bw(16) + facet_wrap(~Symbol) + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))  + ylab(bquote(Expression~(log[2]~CPM))) + ggtitle("CRISPRa")
```


# Initial variancePartition

```{r voom, fig.height=6, fig.width=6}
design = model.matrix( ~ Gene + MDS1 + MDS2, info)
vobj = voom(dge, design, plot=TRUE)
```

variancePartition

```{r vp, eval=TRUE}
vp = fitExtractVarPartModel( vobj, ~ (1|Method) + (1|Gene) + (1|COVID) + (1|Plate) + RIN + SARS.CoV.2 + MDS1 + MDS2, info, BPPARAM=SnowParam(4, progressbar=TRUE))

plotVarPart( sortCols( vp ) ) + theme(aspect.ratio=1)
```

## variancePartition for IL10RB expression
```{r vp.IL10RB, fig.height=3}
plotPercentBars(vp[geneInfo[Symbol=="IL10RB",Geneid],]) + theme(legend.position="right")
```

# Regress out MDS
Removing the first two components reveals clear structure by gene
```{r voom.mds, fig.height=6, fig.width=6}
design = model.matrix( ~ MDS1 + MDS2, info)
vobj = voom(dge, design, plot=FALSE)

fit = dream(vobj, ~ MDS1 + MDS2 + RIN, info)
resid = residuals(fit)

# f = function(fit){
#   fit$model$`responsePlaceholder$E` - get_prediction(fit, ~ 0 + MDS1 + MDS2)
#   }

# resid = fitVarPartModel(vobj, ~ Gene + MDS1 + MDS2, info, fxn=f, BPPARAM = SerialParam())
# resid = do.call(rbind, resid)

C <- cor(t(scale(t(resid))), method="spearman")

hcl = hclust(as.dist(1-C), method='ward.D2')
# plot(hcl)

mds = cmdscale(as.dist(1-C), k=3)

df_mds = data.frame(ID=rownames(mds), MDS1 = mds[,1], MDS2 = mds[,2])

info2 = merge(info, df_mds, by="ID", sort=FALSE)
rownames(info) = info$ID

variables.meta = c('COVID', 'Method', 'Gene', 'Plate')

figList = lapply( variables.meta, function(x){

  ggplot(info2, aes(MDS1.y, MDS2.y)) + geom_point(aes_string(color=x)) + theme_classic() + theme(aspect.ratio=1, plot.title = element_text(hjust = 0.5), legend.position="bottom") + scale_color_discrete( name = x)
})
plot_grid( plotlist = figList, ncol=2)
```

```{r vp.mds, eval=FALSE}
# vp = fitExtractVarPartModel( vobj, ~ (1|Method) + (1|Gene) + (1|COVID) + (1|Plate) + RIN + SARS.CoV.2, info, BPPARAM=SnowParam(4, progressbar=TRUE))

# plotVarPart( sortCols( vp ) ) + theme(aspect.ratio=1)
```

# get residuals just for IL10RB
```{r get.resids}

ensGene = geneInfo[Symbol=="IL10RB",Geneid]

df_IL10RB = data.frame(ID = colnames(resid), IL10RB = resid[ensGene,])
rownames(df_IL10RB) = c()

df = merge(df_IL10RB, info[,1:8], by='ID')

ggplot(df, aes(Gene, IL10RB)) + geom_point()

storeData( 'df_IL10RB', "Residualized expression of IL10RB (MDS1, MDS2, RIN)" )

ensGene = geneInfo[Symbol == "IL10RB", Geneid]
df_IL10RB = data.frame(ID = colnames(vobj), IL10RB = vobj$E[ensGene,])
storeData( 'df_IL10RB', "Observed expression of IL10RB" )

# IFNAR2
ensGene = geneInfo[Symbol=="IFNAR2",Geneid]

df_IFNAR2 = data.frame(ID = colnames(resid), IFNAR2 = resid[ensGene,])
rownames(df_IFNAR2) = c()

df = merge(df_IFNAR2, info[,1:8], by='ID')

ggplot(df, aes(Gene, IFNAR2)) + geom_point()

storeData( 'df_IFNAR2', "Residualized expression of IFNAR2 (MDS1, MDS2, RIN)" )

ensGene = geneInfo[Symbol == "IFNAR2", Geneid]
df_IFNAR2 = data.frame(ID = colnames(vobj), IFNAR2 = vobj$E[ensGene,])
storeData( 'df_IFNAR2', "Observed expression of IFNAR2" )
```




```{r genesets}
gs.kegg = getGenesets(org = "hsa", db = "kegg", gene.id.type = 'ENSEMBL', 
    return.type = "GeneSetCollection")
names(gs.kegg) = sapply(gs.kegg, function(x) paste(setName(x), description(x)))

# res = showAvailableCollections(org="hsa", "enrichr")
# 'COVID-19_Related_Gene_Sets',
# libs = c( 'VirusMINT', 'Virus_Perturbations_from_GEO_down', 'Virus-Host_PPI_P-HIPSTer_2020')
# 'Virus_Perturbations_from_GEO_up')#, 'Virus_Perturbations_from_GEO_down')# , 
libs = 'COVID-19_Related_Gene_Sets'
gs.sets = lapply( libs, function(lib){
  getGenesets(org = "hsa", db = "enrichr", gene.id.type = 'ENSEMBL', 
    return.type = "GeneSetCollection", lib=lib)
})
names(gs.sets) = libs

gs.combine = GeneSetCollection(c(unlist(gs.sets)))

# gs = gs.sets[[3]]

# names(gs) = paste(sapply(gs, description), "- Up")

# gs = lapply(gs.sets[[3]], function(x){
#   setName(x) = paste(names(x), "Up", sep='_')
#   x
# })

gs.GO = get_GeneOntology()

# get GO genesets that contail IL10, IL10RA, IL10RB
gs.GO.symbol = get_GeneOntology(to='SYMBOL')

gs.keep = sapply(gs.GO.symbol, function(x){
  any(geneIds(x) %in% c('IL10', 'IL10RA', 'IL10RB'))
})
go.names.keep = names(gs.GO.symbol)[gs.keep]

# Combine COVID genesets and GO genesets that contail IL10, IL10RA, IL10RB
gsc.test = GeneSetCollection(c(gs.GO[go.names.keep], gs.combine))
```




# Run analysis
```{r define.functions}
run_zenith = function(fit, coefs, gs.collection, n_genes_min=10, n_genes_max=5000 ){
    
  # Map from Ensembl genes in geneSets_GO to 
  # from trimmed Ensembl names from RNA-seq data 
  geneSets.index = ids2indices( gs.collection, trim_ensembl_ids(rownames(fit)))
        
  # filter by size of gene set
  geneSets.index = geneSets.index[sapply(geneSets.index, function(x) (length(x) >= n_genes_min) & (length(x) <= n_genes_max)) ]

  # run zenith for each coefficient
  res = lapply( coefs, function(coef){

    df = zenith(fit, coef, geneSets.index, squaredStats=FALSE)
    df$Set = rownames(df)
    rownames(df) = c()
    df$Coef = coef
    df$FDR = qvalue(df$PValue)$qvalue
    df
    })
  do.call(rbind, res)
}

run_analysis = function(vobj, infoSub, baseline, coefs, form, gsc){

  # Set baseline to 'Scramble CRISPRa'
  levels(infoSub$Gene)
  infoSub$Gene = relevel(factor(infoSub$Gene), baseline)
  # infoSub$Gene = relevel(factor(infoSub$Gene), 'IL10RB-2')
  levels(infoSub$Gene)

  # remove plot because it is colinear with gene
  fit = dream( vobj, form, infoSub)
  fit = eBayes(fit)

  res = run_zenith( fit, coefs, recodeToList(gsc), n_genes_min=20, n_genes_max=1000)
    
  # get t-statistics
  df_tstat = lapply( coefs, function(coef){
    tab = topTable(fit, coef, number=Inf, sort.by='none')
    res = data.frame(Geneid = rownames(tab), coef = tab$t)
    colnames(res)[2] = coef
    res
    })
  df_tstat = Reduce(function(x1, x2) merge(x1, x2, by="Geneid"), df_tstat)

  list(fit    = fit, 
      enrich  = res, 
      df_tstat= df_tstat)
}

gheatmap = function( df_melt, row, col, value, label='', sort.row=TRUE, sort.col=TRUE, zlim=NULL, isSquare=TRUE){

  df_melt[[row]] = factor(df_melt[[row]], unique(df_melt[[row]]))
  df_melt[[col]] = factor(df_melt[[col]],  unique(df_melt[[col]]))

  df = matrix(with(df_melt, delta / se), nlevels(df_melt[[col]]), nlevels(df_melt[[row]]), byrow=TRUE)
  colnames(df) = levels(df_melt[[row]])
  rownames(df) = levels(df_melt[[col]])

  # C = cor(t(df), method="spearman")
  # plotCorrMatrix(C)

  if( sort.col ){
    d.row = as.dist(1-cor(t(df), method="spearman"))
    hcl.row = hclust(d.row, method="ward.D2")
  }else{
    hcl.row = list(order = 1:nrow(df))
  }

  if( sort.row ){
    d.col = as.dist(1-cor(df, method="spearman"))
    hcl.col = hclust(d.col, method="ward.D2")
  }else{
    hcl.col = list(order = 1:ncol(df))
  }

  df_melt[[col]] = factor(df_melt[[col]], rownames(df)[hcl.row$order])
  df_melt[[row]] = factor(df_melt[[row]], colnames(df)[hcl.col$order])

  if( is.null(zlim) ){
    zlim = max(abs(df_melt$value)) 
    zlim = c(-zlim, zlim)
  }
  
  fig =  ggplot(df_melt, aes_string(col, row, fill='value')) + geom_tile() + scale_fill_gradient2(name=label,low="blue", mid="white", high="red", limits=zlim) + theme_classic(8) + theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

  if( isSquare ){
    aspect.ratio = length(unique(df_melt[[row]])) / length(unique(df_melt[[col]]))
    fig = fig + theme(aspect.ratio=aspect.ratio)
  }

  attr(fig, 'row.names') = levels(df_melt[[row]])
  attr(fig, 'col.names') = levels(df_melt[[col]])
  fig
}
```

```{r DE.Test, message=FALSE, fig.height=20}
keep.samples = with(info, Method == "CRISPRa")# & COVID=='CoV-')
keep.samples[] = TRUE
keep <- filterByExpr(geneCounts[,keep.samples])
dge = DGEList( geneCounts[keep,keep.samples])
dge = calcNormFactors(dge)

infoSub = droplevels(info[keep.samples,])

form = ~ Gene + COVID + MDS1 + MDS2 + RIN 
design = model.matrix( form, infoSub)
vobj = voom(dge, design, plot=FALSE)

baseline = 'Scramble CRISPRa'
coefs.CRISPRa = paste0('Gene',c('IL10RB-1', 'IL10RB-2', 'IL10RB-3', 'IL10RB-4'))
# coefs.CRISPRa = c(coefs.CRISPRa, 'GeneScramble shRNA')
res.CRISPRa = run_analysis( vobj, infoSub, baseline, coefs.CRISPRa, form, gsc.test)

baseline = 'Scramble shRNA'
coefs.shRNA = c(paste0('Gene',c('IFNAR2', 'IL10RB')), 'COVIDCoV+')
res.shRNA = run_analysis( vobj, infoSub, baseline, coefs.shRNA, form, gsc.test)

# upload results
df_tstat = merge(res.CRISPRa$df_tstat, res.shRNA$df_tstat, by="Geneid") %>% getGeneSymbol('Geneid')
storeData( 'df_tstat', "t-statistics using scramble as baseline" )
```

Differential expression performed with formula: `r paste(as.character(form), collapse=' ')`

# Show effect of CRISPR on target genes
Compare each IL10RB pertubation to the Scramble CRISPRa baseline, including `~ MDS1 + MDS2 + RIN` as covariates. `expression il10RB ~ variable [perturbation vs baseline] + MDS1 + MDS2 + RIN`.

```{r IL10RB.de}
df_de = lapply(coefs.CRISPRa, function(coef){
  tab = topTable(res.CRISPRa$fit, coef=coef, number=Inf) %>% getGeneSymbol
  data.frame(coef = gsub("Gene", '', coef), tab[tab$Symbol == "IL10RB",c('Symbol','logFC', 't', 'P.Value')])
})
df_de = do.call(rbind, df_de)

df_de[,c('coef', 'logFC', 't', 'P.Value')] %>% kable(caption="Effect of CRISPRa on IL10RB", row.names=FALSE) %>% kable_styling(full_width=FALSE)
```




```{r volcano.IL10RB, fig.width=8, fig.height=10}
res = res.CRISPRa
coefs = coefs.CRISPRa

lapply(coefs, function(x){

  tab = topTable(res$fit, coef=x, number=Inf)
  data.frame(gRNA = gsub("Gene", '', x), Count = sum(tab$adj.P.Val < 0.05))
}) %>% do.call(rbind, .) %>% kable(caption="# DE") %>% kable_styling(full_width=FALSE)

figList = lapply(coefs, function(x){

  tab = topTable(res$fit, coef=x, number=Inf)

  plotVolcano( tab ) + ggtitle(x)
})
plot_grid(plotlist=figList, ncol=2)
```





```{r other.de}
df_de = lapply(coefs.shRNA, function(coef){
  tab = topTable(res.shRNA$fit, coef=coef, number=Inf) %>% getGeneSymbol
  data.frame(coef = gsub("Gene", '', coef), tab[tab$Symbol %in% c("IL10RB", 'IFNAR2'),c('Symbol','logFC', 't', 'P.Value')])
})
df_de = do.call(rbind, df_de)

df_de[with(df_de, coef==Symbol),] %>% dplyr::select(c('coef', 'logFC', 't', 'P.Value')) %>% kable(caption="Effect of shRNA on the target gene", row.names=FALSE) %>% kable_styling(full_width=FALSE)
df_de[with(df_de, coef=='COVIDCoV+'),] %>% dplyr::select(c('coef', 'logFC', 't', 'P.Value')) %>% kable(caption="Effect of SARS-CoV-2 infection", row.names=FALSE) %>% kable_styling(full_width=FALSE)
```



```{r volcano.IFNAR2, fig.width=8, fig.height=10}
res = res.shRNA
coefs = coefs.shRNA


lapply(coefs, function(x){

  tab = topTable(res$fit, coef=x, number=Inf)
  data.frame(gRNA = gsub("Gene", '', x), Count = sum(tab$adj.P.Val < 0.05))
}) %>% do.call(rbind, .) %>% kable(caption="# DE") %>% kable_styling(full_width=FALSE)

figList = lapply(coefs, function(x){

  tab = topTable(res$fit, coef=x, number=Inf)

  plotVolcano( tab ) + ggtitle(x)
})
plot_grid(plotlist=figList, ncol=2)

```




```{r show.zenith.CRISPRa, fig.width=8, fig.height=10}

res = rbind(res.CRISPRa$enrich, res.shRNA$enrich)
gs.any = unique(res[res$FDR < .01,]$Set)
# gs.any = unique(res[res$FDR < .1,]$Set)

resCombine = res[res$Set %in% gs.any,]
resCombine$Coef = factor(gsub("Gene", '', resCombine$Coef))
resCombine$value = with(resCombine, delta/se)
storeData( 'resCombine', "Gene set analysis of differential expression" )
resCombine$isGO = FALSE
resCombine$isGO[grep("GO", resCombine$Set)] = TRUE

# get similarity of gene sets based on jaccard index
df = data.table(resCombine[resCombine$isGO,])
df = df[,unique(data.table(Set, NGenes))]

# get GO
gs = get_GeneOntology()

# get similarity of gene sets based on jaccard index
M = pinnacle::getJaccardMatrix( gs[df$Set] )
res = plotCorrMatrix(M)

# sort genesets by clustering
df$Set = factor(df$Set, colnames(M)[res$colInd])
df = df[match(levels(Set), Set),]

# Get max Jaccard for each geneset
C = M
diag(C) = NA
df_max = data.frame(cormax = apply(C, 1, max, na.rm=TRUE))


df_max = apply(C, 1, function(x){
  i = which.max(x)
  data.frame(ID = names(i), maxCor = x[i])
  })
df_max = do.call(rbind, df_max)

df2 = merge(data.frame(df), df_max, by.x="Set", by.y="row.names")

df2 %>% kable(caption="GO geneset info") %>% kable_styling(full_width=FALSE) 
```


```{r geneset.plots}
exclude = c("GO0034614: cellular response to reactive oxygen species",
"GO0001525: angiogenesis",
"GO0048514: blood vessel morphogenesis",
"GO0001775: cell activation",
"GO0001944: vasculature development",
"GO0002263: cell activation involved in immune response",
"GO0071345: cellular response to cytokine stimulus",
"GO0032943: mononuclear cell proliferation",
"GO0034097: response to cytokine",
"GO0001775: cell activation",
"GO0045765: regulation of angiogenesis",
"GO0045766: positive regulation of angiogenesis",
"GO0045321: leukocyte activation",
"GO0046651: lymphocyte proliferation",
"GO0070301: cellular response to hydrogen peroxide")

newLvls = c(
  'IL10RB-1' = "CRISPRa IL10RB-1",
  'IL10RB-2' = "CRISPRa IL10RB-2",
  'IL10RB-3' = "CRISPRa IL10RB-3",
  'IL10RB-4' = "CRISPRa IL10RB-4",
  'COVIDCoV+'= 'SARS-CoV-2 infection',
  'IL10RB'   = "shRNA IL10RB",
  'IFNAR2'   = "shRNA IFNAR2")

# set zlim
zlim = max(abs(resCombine$value))
zlim = c(-zlim, zlim)

figSigGO = resCombine %>% 
  mutate(Coef = plyr::revalue(Coef, newLvls)) %>% 
  mutate(Coef = factor(Coef, newLvls)) %>% 
  dplyr::filter(isGO) %>% 
  dplyr::filter(!Set %in% exclude) %>% 
  gheatmap('Set', 'Coef', label="t-statistic", sort.col=FALSE, zlim=zlim)
figSigGO
```


```{r geneset.plots2}
figSigOther = resCombine %>% 
  mutate(Coef = plyr::revalue(Coef, newLvls)) %>% 
  mutate(Coef = factor(Coef, newLvls)) %>% 
  dplyr::filter(!isGO) %>% 
  gheatmap('Set', 'Coef', label="t-statistic", sort.col=FALSE, zlim=zlim)
figSigOther
```

```{r cluster.signatures}
# cluster these responses.
df_tstat = merge(res.CRISPRa$df_tstat, res.shRNA$df_tstat, by="Geneid")

C = cor(df_tstat[,-1])

plotCorrMatrix(C)
```



# DE in CoV- samples

```{r DE.Test.covNeg, message=FALSE, fig.height=20}
keep.samples = with(info, COVID=='CoV-')
keep <- filterByExpr(geneCounts[,keep.samples])
dge = DGEList( geneCounts[keep,keep.samples])
dge = calcNormFactors(dge)

infoSub = droplevels(info[keep.samples,])

form = ~ Gene + MDS1 + MDS2 + RIN 
design = model.matrix( form, infoSub)
vobj = voom(dge, design, plot=FALSE)

baseline = 'Scramble CRISPRa'
coefs.CRISPRa = paste0('Gene',c('IL10RB-1', 'IL10RB-2', 'IL10RB-3', 'IL10RB-4'))
# coefs.CRISPRa = c(coefs.CRISPRa, 'GeneScramble shRNA')
res.CRISPRa = run_analysis( vobj, infoSub, baseline, coefs.CRISPRa, form, gsc.test)

baseline = 'Scramble shRNA'
coefs.shRNA = paste0('Gene',c('IFNAR2', 'IL10RB'))
res.shRNA = run_analysis( vobj, infoSub, baseline, coefs.shRNA, form, gsc.test)
```

Differential expression performed with formula: `r paste(as.character(form), collapse=' ')`

# Show effect of CRISPR on target genes
```{r IL10RB.de.covNeg}
df_de = lapply(coefs.CRISPRa, function(coef){
  tab = topTable(res.CRISPRa$fit, coef=coef, number=Inf) %>% getGeneSymbol
  data.frame(coef = gsub("Gene", '', coef), tab[tab$Symbol == "IL10RB",c('Symbol','logFC', 't', 'P.Value')])
})
df_de = do.call(rbind, df_de)

df_de[,c('coef', 'logFC', 't', 'P.Value')] %>% kable(caption="Effect of CRISPRa on IL10RB", row.names=FALSE) %>% kable_styling(full_width=FALSE)
```




```{r volcano.IL10RB.covNeg, fig.width=8, fig.height=10}
res = res.CRISPRa
coefs = coefs.CRISPRa

lapply(coefs, function(x){

  tab = topTable(res$fit, coef=x, number=Inf)
  data.frame(gRNA = gsub("Gene", '', x), Count = sum(tab$adj.P.Val < 0.05))
}) %>% do.call(rbind, .) %>% kable(caption="# DE") %>% kable_styling(full_width=FALSE)

figList = lapply(coefs, function(x){

  tab = topTable(res$fit, coef=x, number=Inf)

  plotVolcano( tab ) + ggtitle(x)
})
plot_grid(plotlist=figList, ncol=2)
```





```{r other.de.covNeg}
df_de = lapply(coefs.shRNA[1:2], function(coef){
  tab = topTable(res.shRNA$fit, coef=coef, number=Inf) %>% getGeneSymbol
  data.frame(coef = gsub("Gene", '', coef), tab[tab$Symbol %in% c("IL10RB", 'IFNAR2'),c('Symbol','logFC', 't', 'P.Value')])
})
df_de = do.call(rbind, df_de)

df_de[with(df_de, coef==Symbol),] %>% dplyr::select(c('coef', 'logFC', 't', 'P.Value')) %>% kable(caption="Effect of shRNA on the target gene", row.names=FALSE) %>% kable_styling(full_width=FALSE)
```



```{r volcano.IFNAR2.covNeg, fig.width=8, fig.height=10}
res = res.shRNA
coefs = coefs.shRNA

lapply(coefs, function(x){

  tab = topTable(res$fit, coef=x, number=Inf)
  data.frame(gRNA = gsub("Gene", '', x), Count = sum(tab$adj.P.Val < 0.05))
}) %>% do.call(rbind, .) %>% kable(caption="# DE") %>% kable_styling(full_width=FALSE)

figList = lapply(coefs, function(x){

  tab = topTable(res$fit, coef=x, number=Inf)

  plotVolcano( tab ) + ggtitle(x)
})
plot_grid(plotlist=figList, ncol=2)
```



```{r show.zenith.CRISPRa.covNeg.previous, fig.width=8, fig.height=10}
res = rbind(res.CRISPRa$enrich, res.shRNA$enrich)

res$Coef = paste0(gsub("Gene", '', res$Coef), " (CoV-)")
res$value = with(res, delta/se)
res2 = rbind(res, resCombine[,colnames(resCombine)!="isGO"])

gs.any = unique(res2[res2$FDR < .01,]$Set)

gheatmap( res2[res2$Set %in% gs.any,], 'Set', 'Coef', label="t-statistic")

storeData( 'res2', "Gene set analysis of differential expression (CoV-)" )
```


```{r show.zenith.CRISPRa.covNeg, fig.width=8, fig.height=10}

res = rbind(res.CRISPRa$enrich, res.shRNA$enrich)
gs.any = unique(res[res$FDR < .05,]$Set)

resCombine2 = res[res$Set %in% gs.any,]
resCombine2$value = with(resCombine2, delta/se)
gheatmap(resCombine2, 'Set', 'Coef', label="t-statistic")
```

```{r zenith.covNEg}

resCombine.covneg = rbind(res.CRISPRa$enrich, res.shRNA$enrich)
resCombine.covneg$Coef = factor(gsub("Gene", '', resCombine.covneg$Coef))
resCombine.covneg$value = with(resCombine.covneg, delta/se)
# storeData( 'resCombine', "Gene set analysis of differential expression" )
resCombine.covneg$isGO = FALSE
resCombine.covneg$isGO[grep("GO", resCombine.covneg$Set)] = TRUE


res = resCombine.covneg[resCombine.covneg$Set %in% attr(figSigGO, "row.names"),]
res$Set = factor( res$Set, attr(figSigGO, "row.names"))

# order based on previous figure
res = res[order(as.numeric(res$Set)),]

res %>% 
  mutate(Coef = plyr::revalue(Coef, newLvls)) %>% 
  mutate(Coef = factor(Coef, newLvls)) %>% 
  dplyr::filter(isGO) %>% 
  dplyr::filter(!Set %in% exclude) %>% 
  gheatmap('Set', 'Coef', label="t-statistic", sort.col=FALSE, 
    sort.row=FALSE, zlim=zlim)
```


```{r zenith.covNEg.not.go}
res = resCombine.covneg[resCombine.covneg$Set %in% attr(figSigOther, "row.names"),]
res$Set = factor( res$Set, attr(figSigOther, "row.names"))

# order based on previous figure
res = res[order(as.numeric(res$Set)),]

res %>% 
  mutate(Coef = plyr::revalue(Coef, newLvls)) %>% 
  mutate(Coef = factor(Coef, newLvls)) %>% 
  dplyr::filter(!isGO) %>% 
  gheatmap('Set', 'Coef', label="t-statistic", sort.col=FALSE, 
    sort.row=FALSE, zlim=zlim)
```

```{r exit1, cache=FALSE}
# rmarkdown::render("analysis.Rmd")

 knitr::knit_exit()
```








```{r DE.Test2, message=FALSE, eval=FALSE}
keep.samples = with(info, COVID=='CoV+')
keep.samples[] = TRUE
keep <- filterByExpr(geneCounts[,keep.samples])
dge = DGEList( geneCounts[keep,keep.samples])
dge = calcNormFactors(dge)

infoSub = droplevels(info[keep.samples,])

design = model.matrix( ~ Gene*COVID + RIN, infoSub)
vobj = voom(dge, design, plot=FALSE)

baseline = 'Scramble CRISPRa'
coefs = paste0('Gene',c('IL10RB-1', 'IL10RB-2', 'IL10RB-3', 'IL10RB-4'), ':COVIDCoV+')
res1 = run_analysis( baseline, coefs, ~ Gene*COVID + RIN)

baseline = 'Scramble shRNA'
coefs = paste0('Gene',c('IFNAR2', 'IL10RB'), ':COVIDCoV+')
res2 = run_analysis( baseline, coefs, ~ Gene*COVID + RIN)

res = rbind(res1$enrich, res2$enrich)

gs.any = unique(res[res$FDR < .1,]$Set)

resCombine = res[res$Set %in% gs.any,]
resCombine$value = with(resCombine, delta/se)
gheatmap(resCombine, 'Set', 'Coef', label="t-statistic")
```


# Differential expression with SARS.CoV.2
While IL10RB expression appears to be correlated with SARS.CoV.2 level, this association goes away 

## Using covariates
```{r test.SARS.CoV.2}
keep.samples = with(info, COVID=='CoV+')# & Method == "CRISPRa")
keep <- filterByExpr(geneCounts[,keep.samples])
dge = DGEList( geneCounts[keep,keep.samples])
dge = calcNormFactors(dge)

infoSub = droplevels(info[keep.samples,])

form = ~ SARS.CoV.2 + MDS1 + MDS2 + RIN 
design = model.matrix(form, infoSub)
vobj = voom(dge, design, plot=FALSE)

baseline = 'Scramble CRISPRa'
coefs.CRISPRa = 'SARS.CoV.2'
res = run_analysis( vobj, infoSub, baseline, coefs.CRISPRa,  form, gsc.test)

tab = topTable(res$fit, coef=coefs.CRISPRa, number=Inf)

plotVolcano( tab ) + ggtitle('SARS.CoV.2 quantification')
```

head(res$enrich)


Using formula: `r paste0(as.character(form), collapse=' ')`

Number of DE genes: `r sum(tab$adj.P.Val < 0.05)`

IL10RB expression is not associated with viral load
```{r test.il10RB}
tab = topTable(res$fit, coef='SARS.CoV.2', number=Inf)

tab = tab %>% getGeneSymbol 
tab %>% filter(Symbol %in% c("IL10RB", 'IFNAR2')) %>% 
  kable(caption="Test IL10RB, IFNAR2 versus viral load") %>% 
  kable_styling(full_width=FALSE)
```

```{r IL10RB_SARS}
ensGene = tab %>% filter(Symbol == "IL10RB") %>% rownames

df = merge(infoSub, data.frame(ID = colnames(vobj), IL10RB.expression =  t(t(vobj$E[ensGene,]))), by="ID")
df$resid = residuals(lm(IL10RB.expression ~ MDS1 + MDS2 + RIN, df))

ggplot(df, aes(resid, SARS.CoV.2)) + geom_point() + geom_smooth(method='lm') + theme_classic() + theme(aspect.ratio=1)

with(df, cor.test(resid, SARS.CoV.2))
```

## no covariates
```{r test.SARS.CoV.2.no_Cov}
form = ~ SARS.CoV.2
design = model.matrix(form, infoSub)
vobj = voom(dge, design, plot=FALSE)

baseline = 'Scramble CRISPRa'
coefs.CRISPRa = 'SARS.CoV.2'
res = run_analysis( vobj, infoSub, baseline, coefs.CRISPRa,  form, gsc.test)

tab = topTable(res$fit, coef=coefs.CRISPRa, number=Inf)

plotVolcano( tab ) + ggtitle('SARS.CoV.2 quantification')
```

Using formula: `r paste0(as.character(form), collapse=' ')`

Number of DE genes: `r sum(tab$adj.P.Val < 0.05)`

IL10RB expression is not associated with viral load
```{r test.il10RB.no_Cov}
tab = topTable(res$fit, coef='SARS.CoV.2', number=Inf)

tab = tab %>% getGeneSymbol 
tab %>% filter(Symbol=="IL10RB") %>% kable(caption="Test IL10RB versus viral load") %>% kable_styling(full_width=FALSE)
```

```{r IL10RB_SARS.no_cov}
ensGene = tab %>% filter(Symbol=="IL10RB") %>% rownames

df = merge(infoSub, data.frame(ID = colnames(vobj), IL10RB.expression =  t(t(vobj$E[ensGene,]))), by="ID")

ggplot(df, aes(IL10RB.expression, SARS.CoV.2)) + geom_point() + geom_smooth(method='lm') + theme_classic() + theme(aspect.ratio=1)

with(df, cor.test(IL10RB.expression, SARS.CoV.2))
```



```{r exit, cache=FALSE}
# rmarkdown::render("analysis.Rmd")
knitr::knit_exit()
```



baseline = 'Scramble shRNA'
coefs.shRNA = 'SARS.CoV.2'
res.shRNA = run_analysis( baseline, coefs.shRNA, ~ Gene + SARS.CoV.2)


head(res.shRNA$enrich)



res = rbind(res.CRISPRa$enrich, res.shRNA$enrich)

gs.any = unique(res[res$FDR < .01,]$Set)

resCombine = res[res$Set %in% gs.any,]
resCombine$value = with(resCombine, delta/se)
gheatmap(resCombine, 'Set', 'Coef', label="t-statistic")






```{r de.dream.shRNA, message=TRUE}
keep.samples = info$Method == "shRNA"
keep <- filterByExpr(geneCounts[,keep.samples])
dge = DGEList( geneCounts[keep,keep.samples])
dge = calcNormFactors(dge)

infoSub = droplevels(info[keep.samples,])

design = model.matrix( ~ Gene + COVID + RIN, infoSub)
vobj = voom(dge, design, plot=TRUE)


# Set baseline to 'Scramble CRISPRa'
levels(infoSub$Gene)
infoSub$Gene = relevel(factor(infoSub$Gene), 'Scramble shRNA')
levels(infoSub$Gene)

# remove plot because it is colinear with gene
fit.IFNAR2 = dream( vobj, ~ Gene + COVID + RIN, infoSub)
fit.IFNAR2 = eBayes(fit.IFNAR2)

gs.GO = get_GeneOntology()

res.IFNAR2 = run_zenith( fit.IFNAR2, "GeneIFNAR2", recodeToList(gs.GO))
```



# tab = topTable(fit.IFNAR2, coef='GeneIFNAR2', number=Inf)
# sum(tab$adj.P.Val < 0.05)


# res = run_zenith( fit.IFNAR2, "COVIDCoV+", recodeToList(gs.GO))





# tab = topTable(fit.CRISPRa, coef='GeneIL10RB-4', number=Inf)
# sum(tab$adj.P.Val < 0.05)


# res = run_zenith( fit.CRISPRa, 'GeneIL10RB-4', recodeToList(gs.GO))


# gs = get_MSigDB(c("H", "C7", "C8"))





coef = 'GeneIL10RB-1'
tab = topTable(fit.CRISPRa, coef=coef, number=Inf)
sum(tab$adj.P.Val < 0.05)

# tab = getGeneSymbol( tab )
# tab[tab$Symbol == "IL10RB",]



res = run_zenith( fit.CRISPRa, colnames(fit.CRISPRa)[2:8], recodeToList(gs.COVID19), n_genes_min=20, n_genes_max=1000)
head(res)

gs.any = unique(res[res$FDR < .1,]$Set)

res2 = res[res$Set %in% gs.any,]

aspect.ratio = length(unique(res2$Set)) / length(unique(res2$Coef))

zlim = with(res2, max(abs(delta/se)))
zlim = c(-zlim, zlim)

ggplot(res2, aes(Coef, Set, fill=delta/se)) + geom_tile() + scale_fill_gradient2(low="blue", mid="white", high="red", limits=zlim) + theme_classic(8) + theme(aspect.ratio=aspect.ratio, plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))












res = run_zenith( fit.CRISPRa, "GeneIL10RB-1", recodeToList(gs.GO), n_genes_min=20, n_genes_max=1000)
head(res)



res = run_zenith( fit.CRISPRa, "GeneIL10RB-1", recodeToList(gs.kegg), n_genes_min=20, n_genes_max=1000)
head(res)




```


res.CRISPRa = run_zenith( fit.CRISPRa, "GeneIL10RB-1", recodeToList(gs.GO))
```

# fit.CRISPRa = lmFit( vobj,model.matrix( ~ Gene + COVID, infoSub))
# fit.CRISPRa = eBayes(fit.CRISPRa)
#   plotVolcano( tab ) 

tab = topTable(fit.CRISPRa, coef='GeneIL10RB-1', number=Inf)
sum(tab$adj.P.Val < 0.05)


res = run_zenith( fit.CRISPRa, "GeneIL10RB-1", recodeToList(gs.GO), n_genes_min=20, n_genes_max=1000)
head(res)






```{r de.dream.ifnar2, message=TRUE}
# Set baseline to 'Scramble CRISPRa'
levels(info$Gene)
info$Gene = relevel(factor(info$Gene), 'Scramble shRNA')
levels(info$Gene)

# remove plot because it is colinear with gene
fit.IFNAR2 = dream( vobj, ~ Gene + COVID + Method, info)
fit.IFNAR2 = eBayes(fit.IFNAR2)

gs.GO = get_GeneOntology()

res.IFNAR2 = run_zenith( fit.IFNAR2, "GeneIFNAR2", recodeToList(gs.GO))
```








# Differential expression
```{r de.dream, message=TRUE}
# Set baseline to 'Scramble CRISPRa'
levels(info$Gene)
info$Gene = relevel(factor(info$Gene), 'Scramble CRISPRa')
levels(info$Gene)

# remove plot because it is colinear with gene
fit = dream( vobj, ~ Gene*COVID, info)
fit = eBayes(fit)

# colnames(fit)
gs.GO = get_GeneOntology()

res = run_zenith( fit, colnames(fit)[2:8], recodeToList(gs.GO))
```



res[grep("GO0002250", res$Set),]


Get p-value for IL10RB gene.  Does it change????



```{r volcano}
figList = lapply(colnames(fit)[3:7], function(x){

  tab = topTable(fit, coef=x, number=Inf)

  plotVolcano( tab ) + ggtitle(x)
})
plot_grid(plotlist=figList, ncol=3)
```

```{r number.DE}
sapply(colnames(fit)[3:7], function(x){

  tab = topTable(fit, coef=x, number=Inf)
  sum(tab$adj.P.Val < 0.05)
}) %>% kable(caption="# DE") %>% kable_styling(full_width=FALSE)
```


table(res$FDR < .1)






# fit contrasts
```{r fit.contrasts}






```

```{r test, eval=FALSE}


rmarkdown::render("analysis.Rmd")

gs.GO = get_GeneOntology()

res = run_zenith( fitd, colnames(fitd)[-1], recodeToList(gs.GO))

fit = lmFit( vobj, model.matrix(~ Gene + COVID + Plate, info))
fit = eBayes(fit)

# colnames(fit)

tab = topTable(fit, coef='GeneIFNAR2', number=Inf)


# gs.MSigDB = get_MSigDB()    



# Compare Scramble CRISPRa vs shRNA
# 4 ILR10RB CRISPRa has variable impact
#   Can I correlate ILR10RB levels with CoV levels or anything else
# 
Show that baseline IL10RB expression is low




```



DE effects in COV- case.
Differential expression with SARS.CoV.2
  look at top genes






















